libraries/chain/include/steem/chain/smt_objects/account_balance_object.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <steem/chain/steem_object_types.hpp>
+#include <steem/protocol/smt_operations.hpp>
+
+#ifdef STEEM_ENABLE_SMT
+
+namespace steem { namespace chain {
+
+/**
+ * Class responsible for holding non STEEM balance (SMT one) for given account.
+ * Since multiple kinds of balances are possible (atm regular and savings) this
+ * class is parameterized to distinguish their types. 
+ */
+template <enum object_type ObjectType>
+class account_balance_object : public object< ObjectType, account_balance_object<ObjectType> >
```

Comment:
I would consider combining the normal balance and savings balance in the same object. It may be a little more space, but it probably makes the implementation a bit easier.

Reply:
This part can be performance (especially memory consumption) sensitive since we are adding separate index for getting basic account data as its balance. Having limited mem. consumption and  make index choosing simply by compile time type-system is best what we can achieve here. The question is how much possible is to have account(s) without SMT savings...